### PR TITLE
chore(build process): Add semantic-release workflow [skip ci]

### PR DESCRIPTION
**Pull request by ianwhitedeveloper**
Add semantic-release Github workflow to repo.
Add @semantic-release/git plugin and configuration to [automatically push release related `package.json` changes to repository](https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/github-actions.md#pushing-packagejson-changes-to-a-master-branch).

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -3,8 +3,15 @@ name: Node_CI
 on: [pull_request,push]
 
 jobs:
-  build:
+  prepare: # Do not run workflow for chore commits https://bit.ly/39qUH60
+    runs-on: ubuntu-latest
+    if: "! contains(github.event.head_commit.message, '[skip ci]')"
```
Comment (tuckbick) [inline .github/workflows/nodejs.yml:8]: Neat!

File: .releaserc.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "plugins": [
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
```
Comment (tuckbick) [inline .releaserc.json:4]: * According to the plugin descriptions [here](https://semantic-release.gitbook.io/semantic-release/extending/plugins-list#official-plugins), these two plugins are already included by default when semantic-release is ran? 🤔 
* Seems like we'd also want to include `@semantic-release/changelog`.

Comment (ianwhitedeveloper) [reply .releaserc.json:4]: Ah cool I hadn't seen that plugins page. I basically copied this configuration from the [Usage directions](https://github.com/semantic-release/git#usage) of the `@semantic-release/git` plugin. Looks like we could also make this [shareable](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/shareable-configurations.md) once we confirm this works.

Good call on the `@semantic-release/changelog` -- I'll read up on how to configure.

**Comment by ianwhitedeveloper** [comment]
Replaced the `npm ci` command I copied over from the example code with `npm install` because `npm ci ` lists this as one of its requirements, which this repo doesn't satisfy:

> The project must have an existing package-lock.json or npm-shrinkwrap.json

**Pull request merged**